bridge-token/src/lib.rs

```diff
@@ -120,6 +125,45 @@ impl BridgeToken {
         let caller = env::predecessor_account_id();
         caller == self.controller || caller == env::current_account_id()
     }
+
+    pub fn is_paused(&self) -> bool {
+        self.paused != 0 && !self.controller_or_self()
+    }
+
+    pub fn set_paused(&mut self, paused: bool) {
+        require!(self.controller_or_self());
+        self.paused = if paused { 1 } else { 0 };
+    }
+
+    pub fn upgrade_and_migrate(&self) {
+        require!(
+            self.controller_or_self(),
+            "Only the controller or self can update the code"
+        );
+
+        // Receive the code directly from the input to avoid the
+        // GAS overhead of deserializing parameters
+        let code = env::input().unwrap_or_else(|| panic!("ERR_NO_INPUT"));
+        // Deploy the contract code.
+        let promise_id = env::promise_batch_create(&env::current_account_id());
+        env::promise_batch_action_deploy_contract(promise_id, &code);
```

Comment:
Does the specification of how much gas it used exist?

Reply:
It depends on the code size, here is more details 
https://docs.near.org/concepts/basics/transactions/gas-advanced